src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -206,6 +203,9 @@
     identify the scenarios in which this property is populated.
     -->
     <Field Name="BoundContainingTypeOpt" Type="BoundTypeExpression" Null="allow"/>
+
+    <!-- Non-null type is required for this node kind -->
```

Comment:
`TypeWithAnnotations` is a struct, so the standard non-null checks the bound node writer generates does not verify this constraint. Consider adding support to the writer to allow it to correctly check `TypeWithAnnotation.Type`, or add some other verification method to ensure that this is still true.

Reply:
Ah, I see you did that in one of the constructors. Still consider adding support in the writer though.

---
In reply to: [267964867](https://github.com/dotnet/roslyn/pull/34273#discussion_r267964867) [](ancestors = 267964867)

Reply:
I don't think any consumer is likely to provide the type twice, which is what would be required to use the auto-generated constructor.

---
In reply to: [267971470](https://github.com/dotnet/roslyn/pull/34273#discussion_r267971470) [](ancestors = 267971470,267964867)